### DISCUSSION TITLE
Input boolean

### DISCUSSION THREAD

**Discussion by AndreKR** [zachowj/node-red-contrib-home-assistant-websocket#1117] [Feature Request]
All the other input helpers (number, text, select, time) are there, but not the [Input Boolean](https://www.home-assistant.io/integrations/input_boolean/).

**Comment by zachowj** [comment]
The switch node is an over complex input boolean

**Reply by AndreKR** [reply]
Wait, is it actually the same thing? I just saw that the Input Boolean is called a "toggle" in the UI and it has a switch icon.

I didn't want to use the switch node because I assumed it will create a switch (the same as for example an actual smart switch creates using MQTT discovery) which seemed semantically wrong for storing binary sensor input. I also couldn't create a binary sensor because I want my Boolean/Toggle to be *set* from Node-RED but to be *reset* from Home Assistant. the Input Boolean seemed very appropriate for this.

**Reply by zachowj** [reply]
Home Assistant doesn't expose an entity or domain, for input boolean type entities, as they do for numbers, text, and selects. The switch domain is the closest.

https://developers.home-assistant.io/docs/core/entity

**Reply by AndreKR** [reply]
Hm, I see. I assumed those nodes were related to the `input_...` entities (like [`input_text`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/input_text/)), but in reality they are related to those "building block integrations" (like [`text`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/text/)).
